### PR TITLE
Add HotRod store for authorization services

### PR DESCRIPTION
**Pull request by mhajas**
PR #9798 needs to be merged before this one.

### CODE REVIEW

**Review by martin-kanis — COMMENTED** [review]
@mhajas Thanks for the PR! It looks good. I have just two comments.

File: model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/IckleQueryMapModelCriteriaBuilder.java [hunk 1]
```diff
@@ -74,6 +84,7 @@
         ANALYZED_MODEL_FIELDS.add(UserModel.SearchableFields.FIRST_NAME);
         ANALYZED_MODEL_FIELDS.add(UserModel.SearchableFields.LAST_NAME);
         ANALYZED_MODEL_FIELDS.add(UserModel.SearchableFields.EMAIL);
+        ANALYZED_MODEL_FIELDS.add(Policy.SearchableFields.TYPE);
```
[the remark this answers is not in the document]
Comment (martin-kanis) [reply model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/IckleQueryMapModelCriteriaBuilder.java:87]: OK it's hidden in MapPolicyStore 
```
case TYPE:
case NAME:
        return mcb.compare(name.getSearchableModelField(), Operator.ILIKE, "%" + value[0] + "%");
```

File: model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/IckleQueryWhereClauses.java [hunk 2]
```diff
@@ -154,4 +156,30 @@ private static String whereClauseForConsentClientFederationLink(String modelFiel
         String providerId = new StorageId((String) values[0], "").getId();
         return IckleQueryOperators.combineExpressions(ModelCriteriaBuilder.Operator.LIKE, getFieldName(UserModel.SearchableFields.CONSENT_FOR_CLIENT), new String[] {providerId + "%"}, parameters);
     }
+
+    private static String whereClauseForPolicyConfig(String modelFieldName, ModelCriteriaBuilder.Operator op, Object[] values, Map<String, Object> parameters) {
+        final Object attrName = values[0];
```
[the remark this answers is not in the document]
Comment (mhajas) [reply model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/IckleQueryWhereClauses.java:183]: We throw an exception on line 180 if `values.length == 0`

File: model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/authorization/HotRodPolicyConfigPair.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+package org.keycloak.models.map.storage.hotRod.authorization;
+
+import org.infinispan.protostream.annotations.ProtoDoc;
+import org.infinispan.protostream.annotations.ProtoField;
+
+import java.util.Objects;
+
+@ProtoDoc("@Indexed")
+public class HotRodPolicyConfigPair {
```
Comment (martin-kanis) [inline model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/authorization/HotRodPolicyConfigPair.java:26]: I think we might leverage more generic `HotRodStringPair` here. WDYT?

**Comment by mhajas** [comment]
This PR will need to wait for this: https://github.com/keycloak/keycloak/pull/11816 to be merged first as it enables hot-rod tests in GHA again. 

Also, there are some test failures in [GHA in my fork](https://github.com/mhajas/keycloak/runs/6290860301?check_suite_focus=true), I will investigate these.

**Review by martin-kanis — APPROVED** [review]
@mhajas Thanks! It looks good. Can you mark it as "ready for review" or it's still in progress?

**Review by martin-kanis — APPROVED** [review]

**Review by hmlnarik — APPROVED** [review]
Approving per @martin-kanis 's approval

**Pull request merged**